### PR TITLE
Machine Learning: Add the core components

### PR DESCRIPTION
**Pull request by bogalvan**
Pls CR :smile:

### CODE REVIEW

File: ontologyToTagMap.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import json
+from pprint import pprint
+
+data= {}
+data[''] = [] 
+
+with open('ontology.json') as f:
+    loaded_json = json.load(f)
+
+
```
Comment (tvillarete) [inline ontologyToTagMap.py]: Looks like there's an extra line here 😢

File: ontologyToTagMap.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+with open('tag_map.json','w') as f:
+      f.write("{\n")
```
Comment (tvillarete) [inline ontologyToTagMap.py]: Is this tabbed in too far?

File: tagger.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    optimizer = nlp.begin_training()
+    for i in range(n_iter):
+        random.shuffle(TRAIN_DATA)
+        losses = {}
+        # batch up the examples using spaCy's minibatch
+        batches = minibatch(TRAIN_DATA, size=compounding(4., 32., 1.001))
+        for batch in batches:
+            texts, annotations = zip(*batch)
+            nlp.update(texts, annotations, sgd=optimizer, losses=losses)
+        #print('Losses', losses)
```
Comment (tvillarete) [inline tagger.py:97]: This looks extraneous, can we delete it?

**Review by tvillarete — COMMENTED** [review]
Looks good!

File: tagger.py [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+            output_dir.mkdir()
+        nlp.to_disk(output_dir)
+        print("Saved model to", output_dir)
+
+        # test the save model
+        print("Loading from", output_dir)
+        nlp2 = spacy.load(output_dir)
+        doc = nlp2(test_text)
+        print('Tags', [(t.text, t.tag_, t.pos_) for t in doc])
+
```
Comment (lawhvictoria) [inline tagger.py]: Extra spacing

File: tagger.py [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+
+# In[4]:
+
+
+@plac.annotations(
+    lang=("ISO Code of language to use", "option", "l", str),
+    output_dir=("Optional output directory", "option", "o", Path),
+    n_iter=("Number of training iterations", "option", "n", int))
+
```
Comment (lawhvictoria) [inline tagger.py:56]: Extra new lines

**Pull request merged**